src/app/runtime_genesis_ledger/runtime_genesis_ledger.ml

```diff
@@ -2,56 +2,84 @@ open Core
 open Async
 module Ledger = Mina_ledger.Ledger
 
-type t = Ledger.t
-
-module Hash_json = struct
-  type epoch_data = { staking_hash : string; next_hash : string option }
+module Hashes = struct
+  type t = { s3_data_hash : string; ledger_hash : string }
```

Comment:
```suggestion
  type t = { s3_data_hash : string; hash : string }
```

This suggestion would be necessary to unlock another suggestion/simplification above.